PrivilegedActionException: Cannot run program "latexindent.pl"

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
PyCharm 2021.1.1 (build PC-211.7142.13)

### Operating System 
Mac OS X 11.2.3 (x86_64)

### TeXiFy IDEA version
0.7.7

### Description


### Stacktrace
```
java.security.PrivilegedActionException: java.security.PrivilegedActionException: com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "latexindent.pl" (in directory "/Users/daniel/PycharmProjects/10K_comp_Network/LaTeX_Thesis/src"): error=2, No such file or directory
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:808)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.security.PrivilegedActionException: com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "latexindent.pl" (in directory "/Users/daniel/PycharmProjects/10K_comp_Network/LaTeX_Thesis/src"): error=2, No such file or directory
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	... 16 more
Caused by: com.intellij.execution.process.ProcessNotCreatedException: Cannot run program "latexindent.pl" (in directory "/Users/daniel/PycharmProjects/10K_comp_Network/LaTeX_Thesis/src"): error=2, No such file or directory
	at com.intellij.execution.configurations.GeneralCommandLine.createProcess(GeneralCommandLine.java:385)
	at nl.hannahsten.texifyidea.action.ReformatWithLatexindent.actionPerformed(ReformatWithLatexindent.kt:52)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAware$5(ActionUtil.java:273)
	at com.intellij.util.SlowOperations.lambda$allowSlowOperations$0(SlowOperations.java:77)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:64)
	at com.intellij.util.SlowOperations.allowSlowOperations(SlowOperations.java:76)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:273)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.lambda$actionPerformed$0(ActionMenuItem.java:310)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:286)
	at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:77)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:299)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.lambda$fireActionPerformed$0(ActionMenuItem.java:110)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:95)
	at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:110)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:401)
	at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:369)
	at java.desktop/com.apple.laf.ScreenMenuItemCheckbox.itemStateChanged(ScreenMenuItemCheckbox.java:198)
	at java.desktop/java.awt.CheckboxMenuItem.processItemEvent(CheckboxMenuItem.java:396)
	at java.desktop/java.awt.CheckboxMenuItem.processEvent(CheckboxMenuItem.java:364)
	at java.desktop/java.awt.MenuComponent.dispatchEventImpl(MenuComponent.java:375)
	at java.desktop/java.awt.MenuComponent.dispatchEvent(MenuComponent.java:363)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:781)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	... 21 more
Caused by: java.io.IOException: Cannot run program "latexindent.pl" (in directory "/Users/daniel/PycharmProjects/10K_comp_Network/LaTeX_Thesis/src"): error=2, No such file or directory
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
	at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
	at com.int

--- a/src/nl/hannahsten/texifyidea/action/DeleteGeneratedFiles.kt
+++ b/src/nl/hannahsten/texifyidea/action/DeleteGeneratedFiles.kt
@@ -1,5 +1,7 @@
 package nl.hannahsten.texifyidea.action
 
+import com.intellij.notification.Notification
+import com.intellij.notification.NotificationType
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.ui.Messages
@@ -9,13 +11,24 @@ import nl.hannahsten.texifyidea.util.getLatexRunConfigurations
 import nl.hannahsten.texifyidea.util.magic.FileMagic
 import nl.hannahsten.texifyidea.util.runWriteAction
 import java.io.File
+import java.io.IOException
+import java.security.PrivilegedActionException
 
 /**
  * Similar to [DeleteAuxFiles].
  */
 class DeleteGeneratedFiles : AnAction() {
 
-    override fun actionPerformed(e: AnActionEvent) {
+    override fun actionPerformed(event: AnActionEvent) {
+        try {
+            deleteFiles(event)
+        }
+        catch (e: PrivilegedActionException) {
+            Notification("LaTeX", "Could not delete some files", e.message ?: "", NotificationType.ERROR).notify(event.project)
+        }
+    }
+
+    private fun deleteFiles(e: AnActionEvent) {
         val project = getEventProject(e) ?: return
         val basePath = project.basePath ?: return
 

--- a/src/nl/hannahsten/texifyidea/action/DeleteGeneratedFiles.kt
+++ b/src/nl/hannahsten/texifyidea/action/DeleteGeneratedFiles.kt
@@ -11,7 +11,6 @@ import nl.hannahsten.texifyidea.util.getLatexRunConfigurations
 import nl.hannahsten.texifyidea.util.magic.FileMagic
 import nl.hannahsten.texifyidea.util.runWriteAction
 import java.io.File
-import java.io.IOException
 import java.security.PrivilegedActionException
 
 /**

--- a/src/nl/hannahsten/texifyidea/action/ReformatWithLatexindent.kt
+++ b/src/nl/hannahsten/texifyidea/action/ReformatWithLatexindent.kt
@@ -1,6 +1,7 @@
 package nl.hannahsten.texifyidea.action
 
 import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.ProcessNotCreatedException
 import com.intellij.notification.Notification
 import com.intellij.notification.NotificationType
 import com.intellij.openapi.actionSystem.AnAction
@@ -46,10 +47,16 @@ class ReformatWithLatexindent : AnAction("Reformat File with Latexindent") {
         val editor = CommonDataKeys.EDITOR.getData(dataContext) ?: return
         val file = PsiDocumentManager.getInstance(project).getPsiFile(editor.document) ?: return
         if (!file.isLatexFile()) return
-        val process = GeneralCommandLine("latexindent.pl", file.name)
-            .withWorkDirectory(file.containingDirectory.virtualFile.path)
-            .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
-            .createProcess()
+        val process = try {
+            GeneralCommandLine("latexindent.pl", file.name)
+                .withWorkDirectory(file.containingDirectory.virtualFile.path)
+                .withParentEnvironmentType(GeneralCommandLine.ParentEnvironmentType.CONSOLE)
+                .createProcess()
+        }
+        catch (e: ProcessNotCreatedException) {
+            Notification("LaTeX", "Could not run latexindent.pl", "Please double check if latexindent.pl is installed correctly: ${e.message}", NotificationType.ERROR).notify(project)
+            return
+        }
         val output = if (process.waitFor(3, TimeUnit.SECONDS)) {
             process.inputStream.bufferedReader().readText().trim() + process.errorStream.bufferedReader().readText().trim()
         }

--- a/src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/LatexUnicodeInspection.kt
+++ b/src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/LatexUnicodeInspection.kt
@@ -167,7 +167,7 @@ class LatexUnicodeInspection : TexifyInspectionBase() {
      * When neither of these steps is successful, the character is too exotic to replace and an
      * appropriate fail message is shown.
      */
-    private class EscapeUnicodeFix internal constructor(private val inMathMode: Boolean) : LocalQuickFix {
+    private class EscapeUnicodeFix(private val inMathMode: Boolean) : LocalQuickFix {
 
         @Nls
         override fun getFamilyName(): String {
@@ -176,8 +176,17 @@ class LatexUnicodeInspection : TexifyInspectionBase() {
 
         override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
             val element = descriptor.psiElement
+            val editor = FileEditorManager.getInstance(project).selectedTextEditor
 
-            val c = descriptor.textRangeInElement.substring(element.text)
+            val c = try {
+                descriptor.textRangeInElement.substring(element.text)
+            }
+            catch (e: IndexOutOfBoundsException) {
+                if (editor != null) {
+                    HintManager.getInstance().showErrorHint(editor, "Character could not be converted")
+                }
+                return
+            }
 
             // Try to find in lookup for special command
             val replacement: String?
@@ -199,7 +208,6 @@ class LatexUnicodeInspection : TexifyInspectionBase() {
             // When no replacement is found, show error message
             val document = PsiDocumentManager.getInstance(project).getDocument(element.containingFile)
             if (replacement == null) {
-                val editor = FileEditorManager.getInstance(project).selectedTextEditor
                 if (editor != null) {
                     HintManager.getInstance().showErrorHint(editor, "Character could not be converted")
                 }

--- a/src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/packages/LatexPackageNameDoesNotMatchFileNameInspection.kt
+++ b/src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/packages/LatexPackageNameDoesNotMatchFileNameInspection.kt
@@ -30,7 +30,7 @@ class LatexPackageNameDoesNotMatchFileNameInspection : TexifyInspectionBase() {
             .filter { it.name == "\\ProvidesPackage" }
 
         for (command in commands) {
-            val providesName = command.requiredParameters.first().split("/").last()
+            val providesName = command.requiredParameters.firstOrNull()?.split("/")?.last()
             val fileName = file.name.removeSuffix(".sty")
             if (fileName != providesName) {
                 descriptors.add(

